### PR TITLE
AG 6548 - Switch Chart API Explorer to generated docs model

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-6548

Switches the Chart API Explorer to using `interfaces.AUTO.json`, `doc-interfaces.AUTO.json` and `chart-api/overrides.json` as the source of truth for Chart options structures.

This should improve the maintainability of the Chart API Explorer, as changes to `agChartOptions.ts` should now automatically be reflected in it.

Bonus:
- Reduced the amount of `api.json` overrides needed by adding alias types with intrinsic rendering behaviours (see `PixelSize`, `CssColor`, `FontStyle`, `FontWeight`, `FontFamily` and `FontSize`).
- Add support for handling union-types with `type` discriminator (can be seen in Bar option, under `series.type`).
- Intrinsically removes a bunch of broken controls that were present in the previous implementation/configuration.

There is some follow-up work which is detailed as part of the comments on https://ag-grid.atlassian.net/browse/AG-6548 - I will work with Rob to work out what is of value to pursue.

### CODE REVIEW

File: charts-packages/ag-charts-community/src/chart/agChartOptions.ts [hunk 1]
```diff
@@ -969,25 +977,25 @@ export interface AgHistogramSeriesOptions extends AgBaseSeriesOptions {
...
     minAngle?: number;
 }
 
 export interface AgPieSeriesFormatterParams {
     readonly datum: any;
-    readonly fill?: string;
-    readonly stroke?: string;
+    readonly fill?: CssColor;
+    readonly stroke?: CssColor;
     readonly strokeWidth: number;
```
Comment (manapeirov) [inline charts-packages/ag-charts-community/src/chart/agChartOptions.ts:989]: should this one be PixelSize too?

File: community-modules/core/src/ts/interfaces/iAgChartOptions.ts [hunk 2]
```diff
@@ -629,8 +637,8 @@ export interface AgSeriesMarker {
 
 export interface AgSeriesMarkerFormatterParams {
     datum: any;
-    fill?: string;
-    stroke?: string;
+    fill?: CssColor;
+    stroke?: CssColor;
     strokeWidth: number;
```
Comment (manapeirov) [inline community-modules/core/src/ts/interfaces/iAgChartOptions.ts:642]: should this be pixelSize?

File: community-modules/core/src/ts/interfaces/iAgChartOptions.ts [hunk 3]
```diff
@@ -795,25 +803,25 @@ export interface AgAreaSeriesOptions extends AgBaseSeriesOptions {
      */
     fills?: string[];
```
Comment (manapeirov) [inline community-modules/core/src/ts/interfaces/iAgChartOptions.ts:804]: CssColor[]?

File: charts-packages/ag-charts-community/src/chart/agChartOptions.ts (hunk 1 above)
Comment (alantreadway) [reply charts-packages/ag-charts-community/src/chart/agChartOptions.ts:989]: Good spot

Comment (alantreadway) [reply charts-packages/ag-charts-community/src/chart/agChartOptions.ts]: Fixed

File: community-modules/core/src/ts/interfaces/iAgChartOptions.ts (hunk 3 above)
Comment (alantreadway) [reply community-modules/core/src/ts/interfaces/iAgChartOptions.ts]: Fixed

File: community-modules/core/src/ts/interfaces/iAgChartOptions.ts (hunk 2 above)
Comment (alantreadway) [reply community-modules/core/src/ts/interfaces/iAgChartOptions.ts]: Fixed

**Comment by alantreadway** [comment]
Fixes pushed @manapeirov.

File: grid-packages/ag-grid-docs/documentation/src/components/charts-api-explorer/Options.tsx [hunk 4]
```diff
@@ -1,28 +1,76 @@
...
+const FONT_WEIGHT_EDITOR_PROPS = {
+    default: 'normal',
+    breakIndex: 4,
+    options: ['normal', 'bold', 'bolder', 'lighter', '100', '200', '300', '400', '500', '600', '700', '800', '900'],
+};
 
-const FunctionDefinition = ({ definition }) => {
-    const { parameters, returnType } = definition;
-    const returnTypeIsObject = typeof returnType === 'object';
+const FONT_STYLE_EDITOR_PROS = {
```
Comment (manapeirov) [inline grid-packages/ag-grid-docs/documentation/src/components/charts-api-explorer/Options.tsx:26]: FONT_STYLE_EDITOR_PRO**P**S?

File: grid-packages/ag-grid-docs/documentation/src/components/charts-api-explorer/Options.tsx [hunk 5]
```diff
@@ -33,143 +81,540 @@ const FunctionDefinition = ({ definition }) => {
...
 
-        return elements;
+    const axesModelDesc = model.properties['axes']?.desc;
+    if (axesModelDesc?.type === 'array' && axesModelDesc.elements.type === 'union') {
+        axesModelDesc.elements.options = axesModelDesc.elements.options.filter(
+            (o) => o.type === 'nested-object' && o.model.properties['type'].desc.tsType.indexOf(axisType) >= 0
+        );
+    }
+
+    console.log(model);
```
Comment (manapeirov) [inline grid-packages/ag-grid-docs/documentation/src/components/charts-api-explorer/Options.tsx:419]: Should we keep this in?

**Comment by manapeirov** [comment]
Looks good to me, just a couple of small comments

File: grid-packages/ag-grid-docs/documentation/src/components/charts-api-explorer/Options.tsx (hunk 4 above)
Comment (alantreadway) [reply grid-packages/ag-grid-docs/documentation/src/components/charts-api-explorer/Options.tsx]: Fixed.

File: grid-packages/ag-grid-docs/documentation/src/components/charts-api-explorer/Options.tsx (hunk 5 above)
Comment (alantreadway) [reply grid-packages/ag-grid-docs/documentation/src/components/charts-api-explorer/Options.tsx:425]: Nice spot - thanks!

**Comment by alantreadway** [comment]
@manapeirov do you want to re-review the tweaks I've just pushed?

File: charts-packages/ag-charts-community/src/chart/agChartOptions.ts [hunk 6]
```diff
@@ -31,6 +31,8 @@ export type AgChartThemeName =
 /** Alias to denote that a value should be a CSS-compliant color string, such as `#FFFFFF` or `rgb(255, 255, 255)` or `white`. */
 export type CssColor = string;
 
+export type Opacity = number;
```
Comment (manapeirov) [inline charts-packages/ag-charts-community/src/chart/agChartOptions.ts:34]: NIT shall we put a comment for this one like the other two?

Comment (alantreadway) [reply charts-packages/ag-charts-community/src/chart/agChartOptions.ts:35]: 🤦  Thanks! Fixed.

**Review by manapeirov — APPROVED** [review]

**Pull request merged**